src/nodes.coffee

```diff
@@ -1532,6 +1532,9 @@ exports.Param = class Param extends Base
         obj = obj.variable
       # * assignments within destructured parameters `{foo:bar}`
       if obj instanceof Assign
+        # ... possibly with a default value
+        if obj.value instanceof Assign
```

Comment:
```
$ coffee -bpe '({a = b = 1}) ->'
(function(arg) {
  var a, b, ref;
  a = (ref = arg.a) != null ? ref : b = 1;
});
```

Ouch. Good catch. But it has nothing to do with this PR. I’ll fix it in another :)